File: 프로그래머스/0/120846. 합성수 찾기/합성수 찾기.js

function isCombNum(n) {
    let cnt = 0;
    for(let i = 1 ; i <= Math.sqrt(n) ; i++){
        if(n % i === 0){
            cnt++;
            if(i !== parseInt(n / i))
                cnt++;
        }
    }
    
    return cnt >= 3;
}

function solution(n) {
    var answer = 0;
    
    for(let i = 4 ; i <= n ; i++)
        if(isCombNum(i))
            answer++;
    
    return answer;
}

File: 프로그래머스/0/120846. 합성수 찾기/합성수 찾기.java
import java.util.stream.IntStream;

class Solution {
    
    public int get_cnt(int n) {
        int cnt = 0;
        
        for(int i = 1 ; i <= (int)Math.sqrt(n) ; i++)
            if(n % i == 0){
                cnt++;
                if(n / i != i)
                    cnt++;
            }
        
        return cnt;
    }
    
    public int solution(int n) {
        
        return (int)IntStream.range(1, n + 1)
            .filter(i -> get_cnt(i) >= 3)
            .count();
        
    }
}